### ISSUE TITLE
4.1.0 RunWriters errors are not properly accounted.

### ISSUE THREAD

**Issue by josvanroosmalen** [marklogic/marklogic-data-hub#1705]
It seems my temporal stuff is not properly setup causing an error when running FlowRunner resulting in 0 new documents written to the final database:

`"error in runWriters", "XDMP-NOFIELD: xdmp.eval(...), -- Field not defined: systemStart"`

**However my jobs entry is reporting FINISHED with everything went fine:**

```
"status": "FINISHED", 
"successfulEvents": 2, 
"failedEvents": 0, 
"successfulBatches": 1, 
"failedBatches": 0
```

**However, the jobOutput contains the stack trace.** 

```
"jobOutput": ["{\"stack\":\"XDMP-NOFIELD: xdmp.eval( ...) ]
```
So I have:

- A status which is OK in the job document
- An Error strack in the job Output in that same job document.
- 0 new documents in the Final database. 

**This is not new to 4.1.0. 4.0.2 had the same issue.**

**Comment by josvanroosmalen** [comment]
Can someone comment on this? This looks to me as a really serious issue in the accounting of DHF.

**Comment by josvanroosmalen** [comment]
Increase priority and get 4.1.1 shipped. Reproduction project shared via datahub-sig.

**Comment by akshaysonvane** [comment]
Tracking via https://project.marklogic.com/jira/browse/DHFPROD-1667

**Comment by aebadirad** [comment]
This should now be fixed (or at least vastly improved for both sjs and xqy batch error writing) in the develop line, will go out with the next release

**Comment by bsrikan** [comment]
This particular bug is fixed now in develop and will be out next release. This does not fix #1721 though, which still needs to be taken care of.

**Comment by aebadirad** [comment]
Fixed in version 4.2.0
